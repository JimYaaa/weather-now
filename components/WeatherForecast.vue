<script lang="ts" setup>
import type { DailyWeather } from '~~/composables/useWeather'

const props = defineProps({
    weathers: {
        required: true,
        type: Array<DailyWeather>
    },
})
</script>

<template>
    <div class="p-6 color-white backdrop-blur rounded-5">
        <p class="text-3">5-Day Forecast</p>

        <div class="w-full h-1px my-4 bg-white"></div>

        <ul class="min-w-400px flex flex-col justify-between items-center">
            <li 
                v-for="weather, index in props.weathers"
                :key="weather.time"
                class="w-full flex justify-between items-center my-2 last:mb-0 first:mt-0"
            >
                <p class="w-50px">{{ index === 0 ? 'Today' : weather.time }}</p>

                <Icon class="color-white mx-4" :name="weather.weather" size="35" />

                <div class="flex items-center justify-between flex-1">
                    <p class="font-sans font-bold">{{ weather.temperature_min }}</p>
                    <div class="flex-1 h-2px mx-4 bg-white rounded"></div>
                    <p class="font-sans font-bold">{{ weather.temperature_max }}</p>
                </div>
            </li>
        </ul>
    </div>
</template>