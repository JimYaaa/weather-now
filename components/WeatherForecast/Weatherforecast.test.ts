import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, it, expect } from 'vitest'
import WeatherForecast from './Index.vue'

describe('WeatherForecast', () => {
    const propsConfig = {
        weathers: [
            {
                temperature_celsius_max: 30,
                temperature_celsius_min: 25,
                temperature_celsius_unit: '째C',
                temperature_fahrenheit_max: 89,
                temperature_fahrenheit_min: 60,
                temperature_fahrenheit_unit: '째F',
                time: 'Sat',
                weather: 'Sunny'
            }
        ],
    }

    it('when temperature unit is celsius', async () => {
        const component = await mountSuspended(WeatherForecast, {
            props: {
                weathers: propsConfig.weathers,
                temperatureUnit: 'celsius'
            },
        })

        const { weathers } = propsConfig
        const temperature = component.vm.getTemperatureByUnit(weathers[0])
        const weather = weathers[0]

        expect(component.html()).toContain('째C')
        expect(temperature).toMatchObject({
            max: weather.temperature_celsius_max,
            min: weather.temperature_celsius_min,
            unit: weather.temperature_celsius_unit
        })
        expect(component.html()).toMatchSnapshot()
    })

    it('when temperature unit is fahrenheit', async () => {
        const component = await mountSuspended(WeatherForecast, {
            props: {
                weathers: propsConfig.weathers,
                temperatureUnit: 'fahrenheit'
            }
        })

        const { weathers } = propsConfig
        const temperature = component.vm.getTemperatureByUnit(weathers[0])
        const weather = weathers[0]

        expect(component.html()).toContain('째F')
        expect(temperature).toMatchObject({
            max: weather.temperature_fahrenheit_max,
            min: weather.temperature_fahrenheit_min,
            unit: weather.temperature_fahrenheit_unit
        })
        expect(component.html()).toMatchSnapshot()
    })
})