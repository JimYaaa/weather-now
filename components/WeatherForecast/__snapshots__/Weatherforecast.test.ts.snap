// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeatherForecast > when temperature unit is celsius 1`] = `
"<div class="p-6 color-white backdrop-blur-lg rounded-5">
  <p class="text-3">5-Day Forecast</p>
  <div class="w-full h-1px my-4 bg-white"></div>
  <ul class="lg-min-w-400px flex flex-col justify-between items-center">
    <li class="w-full flex justify-between items-center my-2 last:mb-0 first:mt-0">
      <p class="w-50px">Today</p><span class="iconify i-Sunny color-white mx-4" aria-hidden="true" style="font-size: 35px;"></span>
      <div class="flex items-center justify-between flex-1">
        <div class="flex items-start min-w-45px color-white font-sans">
          <p class="text-4 font-sans font-bold">25</p>
          <p class="text-3 font-sans font-bold">째C</p>
        </div>
        <div class="flex-1 h-2px mx-4 bg-white rounded"></div>
        <div class="flex items-start min-w-45px color-white font-sans">
          <p class="text-4 font-sans font-bold">30</p>
          <p class="text-3 font-sans font-bold">째C</p>
        </div>
      </div>
    </li>
  </ul>
</div>"
`;

exports[`WeatherForecast > when temperature unit is fahrenheit 1`] = `
"<div class="p-6 color-white backdrop-blur-lg rounded-5">
  <p class="text-3">5-Day Forecast</p>
  <div class="w-full h-1px my-4 bg-white"></div>
  <ul class="lg-min-w-400px flex flex-col justify-between items-center">
    <li class="w-full flex justify-between items-center my-2 last:mb-0 first:mt-0">
      <p class="w-50px">Today</p><span class="iconify i-Sunny color-white mx-4" aria-hidden="true" style="font-size: 35px;"></span>
      <div class="flex items-center justify-between flex-1">
        <div class="flex items-start min-w-45px color-white font-sans">
          <p class="text-4 font-sans font-bold">60</p>
          <p class="text-3 font-sans font-bold">째F</p>
        </div>
        <div class="flex-1 h-2px mx-4 bg-white rounded"></div>
        <div class="flex items-start min-w-45px color-white font-sans">
          <p class="text-4 font-sans font-bold">89</p>
          <p class="text-3 font-sans font-bold">째F</p>
        </div>
      </div>
    </li>
  </ul>
</div>"
`;
